File: dotnet/src/Functions/CreateOrder.cs
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Azure.WebJobs;
using Microsoft.Azure.WebJobs.Extensions.Http;
using Microsoft.Extensions.Logging;
using MongoEncryption.Util;
using System;
using System.IO;
using System.Threading.Tasks;

namespace MongoEncryption.Functions
{
    public static class CreateOrder
    {
        [FunctionName("CreateOrder")]
        public static async Task<IActionResult> Run(
            [HttpTrigger(AuthorizationLevel.Function, "post", Route = "orders/create")] HttpRequest req,
            ILogger log)
        {
            try
            {
                string requestBody = await new StreamReader(req.Body).ReadToEndAsync();

                await Order.InsertOrderAsync(requestBody);

                return new AcceptedResult();
            }
            catch (Exception ex)
            {
                log.LogError(ex, "Order creation error");
                return new BadRequestResult();
            }
        }
    }
}

File: java/src/main/java/com/function/CreateOrder.java
package com.function;

import java.util.Optional;

import com.function.util.Order;
import com.microsoft.azure.functions.ExecutionContext;
import com.microsoft.azure.functions.HttpMethod;
import com.microsoft.azure.functions.HttpRequestMessage;
import com.microsoft.azure.functions.HttpResponseMessage;
import com.microsoft.azure.functions.HttpStatus;
import com.microsoft.azure.functions.annotation.AuthorizationLevel;
import com.microsoft.azure.functions.annotation.FunctionName;
import com.microsoft.azure.functions.annotation.HttpTrigger;

public class CreateOrder {

    @FunctionName("CreateOrder")
    public HttpResponseMessage run(
            @HttpTrigger(name = "req", 
                methods = {HttpMethod.POST}, 
                authLevel = AuthorizationLevel.FUNCTION,
                route = "orders/create") 
            HttpRequestMessage<Optional<String>> request,
            final ExecutionContext context) {
        
        try
        {
            String requestBody = request.getBody().get();

            Order.InsertOrder(requestBody);

            return request.createResponseBuilder(HttpStatus.ACCEPTED).build();
        }
        catch (Exception ex)
        {
            context.getLogger().warning("Order creation error. Error: " + ex.getMessage());
            return request.createResponseBuilder(HttpStatus.BAD_REQUEST).build();
        }
    }
}